lib/rules/force-types-on-object-props.js

```diff
@@ -0,0 +1,163 @@
+      return bVal(aVal)
+    }
+    return bVal == null || /^[bns]/.test(typeof bVal)
+      ? bVal === aVal
+      : isLooksLike(aVal, bVal)
+  })
+
+/**
+ * @param {ComponentProp} property
+ * @param {RuleContext} context
+ */
+const checkProperty = (property, context) => {
+  if (!property.value) {
+    return
+  }
+
+  if (
+    isLooksLike(property.value, { type: 'Identifier', name: 'Object' }) &&
+    property.node.value.type !== 'TSAsExpression'
+  ) {
+    context.report({
+      node: property.node,
+      message: 'Expected type annotation on object prop.'
+    })
+  }
+
+  if (
+    property.type === 'object' &&
+    property.value.type === 'ObjectExpression' &&
+    property.node.value.type === 'ObjectExpression'
+  ) {
+    const typePropert = property.node.value.properties.find(
+      (prop) =>
+        prop.type === 'Property' &&
+        prop.key.type === 'Identifier' &&
+        prop.key.name === 'type'
+    )
+    if (
+      typePropert &&
+      typePropert.type === 'Property' &&
+      isLooksLike(typePropert.value, { type: 'Identifier', name: 'Object' })
+    ) {
+      context.report({
+        node: property.node,
+        message: 'Expected type annotation on object prop.'
+      })
+    }
+  }
+
+  if (property.node.value.type === 'ObjectExpression') {
+    for (const prop of property.node.value.properties) {
+      if (prop.type !== 'Property') {
+        continue
+      }
+      if (prop.key.type !== 'Identifier' || prop.key.name !== 'type') {
+        continue
+      }
+      if (prop.value.type !== 'TSAsExpression') {
+        continue
+      }
+
+      const { typeAnnotation } = prop.value
+      if (
+        ['TSAnyKeyword', 'TSUnknownKeyword'].includes(typeAnnotation.type) ||
+        !typeAnnotation.typeName ||
+        !['Prop', 'PropType'].includes(typeAnnotation.typeName.name)
+      ) {
+        context.report({
+          node: property.node,
+          message: 'Expected type annotation on object prop.'
+        })
+      }
+    }
+  }
+
+  if (property.node.value.type === 'TSAsExpression') {
+    const { typeAnnotation } = property.node.value
+    if (typeAnnotation.type === 'TSFunctionType') {
+      return
+    }
+    if (
+      [
+        'TSAnyKeyword',
+        'TSTypeLiteral',
+        'TSUnknownKeyword',
+        'TSObjectKeyword'
+      ].includes(typeAnnotation.type) ||
+      !typeAnnotation.typeName ||
+      !['Prop', 'PropType'].includes(typeAnnotation.typeName.name)
+    ) {
+      context.report({
+        node: property.node,
+        message: 'Expected type annotation on object prop.'
+      })
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce adding type declarations to object props',
+      categories: ['base'],
```

Comment:
Again, this should be a `suggestion`.

```suggestion
      categories: ['suggestion'],
```

Please regenerate the docs after changing this.